syftr/evaluation/factory.py

```diff
@@ -0,0 +1,76 @@
+import json
+import re
+import typing as T
+
+from llama_index.core.evaluation.base import BaseEvaluator
+from llama_index.core.evaluation.correctness import CorrectnessEvaluator
+from llama_index.core.prompts import (
+    ChatMessage,
+    ChatPromptTemplate,
+    MessageRole,
+)
+
+from syftr.llm import get_llm
+from syftr.logger import logger
+from syftr.studies import Evaluation
+
+
+def json_parser_function(response: str) -> T.Tuple[T.Optional[float], T.Optional[str]]:
+    json_pattern = r"\{.*\}"
+    match = re.search(json_pattern, response)
+    if match:
+        json_str = match.group(0)
```

Comment:
Find last match, as discussed. 

Add some unit tests for this parser to make sure it properly handles nested JSON, multiple jsons, dangling braces, etc.

Reply:
Done